File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] res = new int[2];
        
        int l = 0;
        int r = numbers.length - 1;
        while (l < r) {
            int nr = numbers[r];
            int nl = numbers[l];
            if (nr + nl == target) {
                res[1] = r + 1;
                res[0] = l + 1;
                break;
            } else if (nr + nl > target) {
                r--;
            } else {
                l++;
            }
            
        }

        return res;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        n = len(nums)
        res, sol = [], []
        cur = 0
        
        def backtrack(i, total):
            if i >= n or total > target:
                return
            if total == target:
                res.append(sol[:])
                return
            sol.append(nums[i])
            
            backtrack(i, total + nums[i])
            sol.pop()
            backtrack(i+1, total)


        backtrack(0, 0)

        return res